forest/bonsai/simulate_gps_data.py

```diff
@@ -157,12 +158,511 @@ def bounding_box(center: Tuple[float, float], radius: int) -> Tuple:
+            day_now: int, day of the week
+        Returns:
+            Action dataclass
+        """
+        time_now = t_s % (24 * 60 * 60)
+
+        if time_now == 0:
+            # if it is a weekday and working/studying
+            # wake up between 8am and 9am
+            if day_now < 5 and self.attributes.main_occupation != "":
+                return Action(
+                    "p",
+                    self.home_coordinates,
+                    [8 * 3600, 9 * 3600],
+                    "home_morning",
+                )
+            # else wake up between 8am and 12pm
+            return Action(
+                "p",
+                self.home_coordinates,
+                [8 * 3600, 12 * 3600],
+                "home_morning",
+            )
+
+        # if haven't yet been to office today
+        if not self.office_today:
+            if update:
+                self.office_today = not self.office_today
+            # if today is office day go to office
+            # work for 7 to 9 hours
+            if day_now in self.office_days:
+                return Action(
+                    "fpf",
+                    self.office_coordinates,
+                    [7 * 3600, 9 * 3600],
+                    "office",
+                )
+            # if today is not office day
+            # work for 7 to 9 hours from home
+            elif day_now < 5:
+                return Action(
+                    "p",
+                    self.home_coordinates,
+                    [7 * 3600, 9 * 3600],
+                    "office_home",
+                )
+
+        # otherwise choose to do something in the free time
+        preferred_exit, location = self.choose_preferred_exit(t_s, update)
+
+        # if chosen to stay home
+        if preferred_exit == "home":
+            # if after 10pm and chosen to stay home
+            # stay for the night until next day
+            if time_now + 2 * 3600 > 24 * 3600 - 1:
+                return Action(
+                    "p_night",
+                    self.home_coordinates,
+                    [24 * 3600 - time_now, 24 * 3600 - time_now],
+                    "home_night",
+                )
+            # otherwise stay for half an hour to 2 hours and then decide again
+            return Action(
+                "p",
+                self.home_coordinates,
+                [0.5 * 3600, 2 * 3600],
+                preferred_exit,
+            )
+        # if deciding to stay at home for the night
+        elif preferred_exit == "home_night":
+            return Action(
+                "p_night",
+                self.home_coordinates,
+                [24 * 3600 - time_now, 24 * 3600 - time_now],
+                preferred_exit,
+            )
+        # otherwise go to the location specified
+        # spend from half an hour to 2.5 hours depending
+        # on active status
+        return Action(
+            "fpf",
+            location,
```

Comment:
From mypy: Argument 2 to "Action" has incompatible type "Tuple[Any, ...]"; expected "Tuple[float, float]"